apidef/api_definitions.go

```diff
@@ -374,6 +374,7 @@ type APIDefinition struct {
 		AllowedAccessTypes     []osin.AccessRequestType    `bson:"allowed_access_types" json:"allowed_access_types"`
 		AllowedAuthorizeTypes  []osin.AuthorizeRequestType `bson:"allowed_authorize_types" json:"allowed_authorize_types"`
 		AuthorizeLoginRedirect string                      `bson:"auth_login_redirect" json:"auth_login_redirect"`
+		RequirePKCE            bool                        `bson:"require_pkce" json:"require_pkce"`
```

Comment:
You should not "Require" PKCE. This is entirely optional & a client's decision to optionally send a `code_challenge`.

Reply:
The language can be changed sure - but in terms of Tyk someone enabling PKCE is the same as requiring it for clients accessing a certain API.

Reply:
I picked require because the RFC talked a lot about servers requiring the PKCE flow not just enabling it

Reply:
For it to be optional you simply dont use the RequirePKCE option and you can send optional code challenges